### PR TITLE
Ignore chunk/filter options upon creating scalar variables.

### PR DESCRIPTION
**Pull request by mraspaud**
this is to address issue #12

### CODE REVIEW

**Comment by shoyer** [comment]
Nice! Could you please add a test to verify this fix? Somewhere in here: https://github.com/shoyer/h5netcdf/blob/master/h5netcdf/tests/test_h5netcdf.py

**Comment by mraspaud** [comment]
That should do it :)

**Comment by shoyer** [comment]
I'm happy to add this to the legacy API for compatibility with netCDF4-python, but I'd rather not add this to the new API, which tries to be entirely consistent with h5py.

File: h5netcdf/legacyapi.py [hunk 1]
```diff
@@ -55,6 +55,12 @@ def _group_cls(self):
     def createVariable(self, varname, datatype, dimensions=(), zlib=False,
                        complevel=4, shuffle=True, fletcher32=False,
                        chunksizes=None, fill_value=None):
+        if len(dimensions) == 0:  # it's a scalar
+            # rip off chunk and filter options.
```
Comment (shoyer) [inline h5netcdf/legacyapi.py]: add "for consistency with netCDF4-python"

File: h5netcdf/tests/test_h5netcdf.py [hunk 2]
```diff
@@ -65,6 +65,9 @@ def write_legacy_netcdf(tmp_netcdf, write_module):
     v = ds.createVariable('scalar', np.float32, ())
     v[...] = 2.0
 
+    v = ds.createVariable('intscalar', np.int64, (), zlib=6, fill_value=None)
```
Comment (shoyer) [inline h5netcdf/tests/test_h5netcdf.py]: a brief comment explaining why this test exists would be helpful

**Comment by mraspaud** [comment]
I added the check to the new api also in order for the tests to be consistent and pass. But I'll just create 'intscalar' without any parameters in the new api tests then.

**Comment by shoyer** [comment]
thanks!

**Pull request merged**